av/audio/frame.pyx

```diff
@@ -130,8 +131,16 @@ cdef class AudioFrame(Frame):
         # map avcodec type to numpy type
         try:
             dtype = np.dtype({
+                'u8'  : 'u1',
+                'u8p' : 'u1',
+                's16' :'<i2',
                 's16p':'<i2',
+                's32' :'<i4',
+                's32p':'<i4',
+                'flt' :'<f4',
                 'fltp':'<f4',
+                'dbl' :'<f8',
```

Comment:
I haven't really dug into it, but I suspect that they can because of the numpy stacking thing. I did have questions there about adopting a strict `shape` protocol.

E.g. if we have 100 audio samples in stereo, planar would have `shape == (2, 100)`, but would non-planar have `shape == (100, 2)` or `(200, )`?

I feel like it should be `(100, 2)`; it is always 2-D, and the channel layout is always expressed in the array shape.

Reply:
This has been dealt with in a more thorough fashion in #422, including the reciprocal `from_ndarray`